### DISCUSSION TITLE
Is there any way to configure Cloudflare such that GH Pages certificate renewal will succeed?

### DISCUSSION THREAD

**Discussion by lukewestby** [community/community#23632] [Other Feature Feedback, Questions, & Ideas]
<p>Here is my situation:</p>
<ul>
<li>I have a GH pages site <code>github.com/myorg/mysite</code> with a custom domain <code>foo.example.com</code>
</li>
<li>The DNS for that site is configured in Cloudflare as a CNAME record from <code>foo.example.com</code> to <code>myorg.github.io</code>
</li>
<li>This CNAME record is behind Cloudflare’s proxy so that we can do some things with caching</li>
<li>The first time we set up the custom domain, we left the Cloudflare proxy <em>off</em> until GitHub had finished provisioning a TLS cert</li>
<li>Once GitHub generated a cert we then turned on the Cloudflare proxy and enabled Full (Strict) encryption</li>
<li>This worked fine for 3 months, but then GitHub attempted to renew its own cert. This failed because <em>something</em> Cloudflare is doing between the internet and GitHub’s server is causing renewal to fail.</li>
</ul>
<p>I’m not sure whether GitHub expects the CNAME record to resolve without Cloudflare’s proxy in place, or if there are certain Let’s Encrypt-related endpoints we can exempt from our page rules etc. that would allow renewal to work.</p>
<p>Has anyone had success dealing with this issue?</p>

**Comment by tcbyrd** [comment, marked as answer]
<blockquote>
<p>I’m not sure whether GitHub expects the CNAME record to resolve without Cloudflare’s proxy in place, or if there are certain Let’s Encrypt-related endpoints we can exempt from our page rules etc. that would allow renewal to work.</p>
</blockquote>
<p>The former. With the proxy feature turned on, effectively you need to handle TLS certificates at Cloudflare, since the IP that comes back in the DNS query belongs to Cloudflare, not GitHub. This means GitHub can’t successfully obtain a certificate. Cloudflare is able to automatically obtain a certificate for you as well and it works basically the same way, but keep in mind that the GitHub UI will not reflect that it’s on HTTPS since Cloudflare is the one enforcing those settings. If you disable Cloudflare’s proxy setting and have it do <em>just</em> DNS resolution, GitHub will be able to obtain and enforce the certificate, but you’ll lose some of the features of Cloudflare that require it to be setup as a proxy.</p>

**Reply by MakotoE** [reply]
I assume that if Cloudflare handles the certificate, the proxy-to-origin connection must be unsecured, right? This would be undesireable and I would rather turn off the proxy.

**Reply by tcbyrd** [reply]
If you treat the origin as `https://<username>.github.io` it can still be https from cloudflare to us. The certificate for that is a wildcard cert we manage for all of `*.github.io` and doesn't need to be provisioned dynamically. As far as I know this is possible with Cloudflare, although I think it would require more setup than just toggling the orange cloud. You effectively need to tell cloudflare "for my domain, proxy the origin request to `https://<username>.github.io`".

**Comment by Jip-Hop** [comment]
<p>I’m expecting exactly the same situation. I have a working SSL certificate from GitHub Pages behind Cloudflare now, but I don’t think it will be able to renew itself when it expires. I was hoping there’d be a page rule to configure which would allow renewal. I’ve disabled “Always Use HTTPS” and in combination with page rules it’s possible to renew Let’s Encrypt certificates behind Cloudflare:</p>
<h2>Page rule 1</h2>
<pre><code class="lang-nohighlight">*debeer.it/.well-known*
SSL: Off
</code></pre>
<h2>Page rule 2</h2>
<pre><code class="lang-nohighlight">*debeer.it/*
Always Use HTTPS
</code></pre>
<p>However this doesn’t work with GitHub Pages. Have you found a way to actually renew the GitHub Pages certificate? I would prefer not to disable the Cloudflare proxy… How about just using “SSL Full” instead of “SSL Full (strict)”? I expect it would continue to work, even after the certificate expires.</p>
